src/commonMain/kotlin/org/hisp/dhis/rules/api/RuleEngine.kt

```diff
@@ -0,0 +1,18 @@
+package org.hisp.dhis.rules.api
+
+import org.hisp.dhis.rules.engine.DefaultRuleEngine
+import org.hisp.dhis.rules.models.*
+
+interface RuleEngine {
+    fun validate(expression: String, dataItemStore: Map<String, DataItem>): RuleValidationResult
+    fun validateDataFieldExpression(expression: String, dataItemStore: Map<String, DataItem>): RuleValidationResult
+    fun evaluateAll(enrollmentTarget: RuleEnrollment?, eventsTarget: List<RuleEvent>, executionContext: RuleEngineContext): List<RuleEffects>
+    fun evaluate(target: RuleEnrollment, ruleEvents: List<RuleEvent>, executionContext: RuleEngineContext): List<RuleEffect>
+    fun evaluate(target: RuleEvent, ruleEnrollment: RuleEnrollment?, ruleEvents: List<RuleEvent>, executionContext: RuleEngineContext): List<RuleEffect>
+
+    companion object {
+        fun getInstance(): RuleEngine {
```

Comment:
If you want to avoid using `RuleEngine.Companion.getInstance());` in Java, you can annotate this method with `@JvmStatic` so you can directly use `RuleEngine.getInstance());` in Java.